src/SpaceService.ts

```diff
@@ -202,7 +202,10 @@ class SpaceService extends Base {
     const json = await throwExpected(
       () => got.get(url),
       // TODO: differentiate between space and cluster not being found
-      { 404: () => new SpaceNotFoundException(name) }
+      {
+        404: () => new SpaceNotFoundException(name),
```

Comment:
We also need to differentiate between cases where we have 404 because there is no space, and when there is no cluster. 

```suggestion
        404: e => e?.body.message.includes(name) ?
          new SpaceNotFoundException(name) :
          new NotFoundException(`Cluster id:${clusterId} not found in space '${name}'`),
```

Reply:
Just since we're already looking around here

Reply:
I don't want to make extra changes in this PR, I just need this 403 so I can update the sdk in extension (which is high priority change).